src/models/data.cpp

```diff
@@ -292,6 +292,42 @@ TopDownResults TopDownResults::fromBottomUp(const BottomUpResults& bottomUpData)
     return results;
 }
 
+PerLibraryResults PerLibraryResults::fromTopDown(const TopDownResults& topDownData)
+{
+    PerLibraryResults results;
+    QHash<QString, int> binaryToResultIndex;
+    results.costs.initializeCostsFrom(topDownData.selfCosts);
+
+    std::function<void(const TopDown& parent)> topDownIterator = [&topDownIterator, &results, &binaryToResultIndex,
```

Comment:
please use a separate free function instead of a recursive lambda - the type erasure in std::function requires a memory allocation that's simply not needed here. pass the required data as arguments instead